src/index.ml

```diff
@@ -235,9 +236,13 @@ struct
     | Some log ->
         let offset = IO.offset log.io in
         let h = IO.Header.get log.io in
-        (* If the generation has changed, reload everything. *)
-        if t.generation <> h.generation then sync_log_entries log
-          (* else if the disk offset is greater, reload the newest data. *)
+        (* If the generation has changed *)
+        if t.generation <> h.generation then (
+          (* close the file .*)
+          IO.close log.io;
+          (* check that file is on disk, reopen and reload everything. *)
+          t.log_async <- try_load_log t (Layout.log_async ~root:t.root)
```

Comment:
For the record: this is likely to be slow when there is a blocked merge but this is similar to what was done before (e.g #296) ; we'll work on a fix in a separate PR.